Tools/TestWebKitAPI/Tests/WebKitCocoa/AdditionalReadAccessAllowedURLs.mm

```diff
@@ -33,13 +33,32 @@
 #import "PlatformUtilities.h"
 #import "Utilities.h"
 #import "WKWebViewConfigurationExtras.h"
+#import <WebKit/NSAttributedStringPrivate.h>
 #import <WebKit/WKProcessPoolPrivate.h>
 #import <WebKit/WKWebViewPrivate.h>
 #import <WebKit/_WKProcessPoolConfiguration.h>
 #import <WebKit/_WKRemoteObjectInterface.h>
 #import <WebKit/_WKRemoteObjectRegistry.h>
 #import <wtf/RetainPtr.h>
 
+static std::pair<NSURL *, NSURL *> readableAndUnreadableDirectories()
```

Comment:
The `std::pair<>` return values are bare Objective-C pointers, so some times when the test runs, they'll probably cause crashes.

The return type should probably be:  `std::pair<RetainPtr<NSURL>, RetainPtr<NSURL>>`

Reply:
It is not true that we we need `RetainPtr` here. Autorelease works just as well on the pair as it would work if the result was just an `NSURL`.

We can and should start using ARC when we write test code, making it much easier to get this right and avoiding any debate about this.

Reply:
Fixed!